### PR TITLE
feat: set up scaffholding

### LINKED ISSUE #1: Set up default layout for lab

### PR DESCRIPTION
**Pull request by pyphilia**
Screenshot 2020-10-28 at 16 18 21
Screenshot 2020-10-28 at 16 18 31
Screenshot 2020-10-28 at 16 18 02
Screenshot 2020-10-28 at 16 17 52

Screenshot 2020-10-28 at 16 20 40

close #1

### CODE REVIEW

File: src/actions/layout.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
-const openSettings = () => (dispatch) =>
+const toggleSettings = (showSettings) => (dispatch) =>
   dispatch({
-    type: OPEN_SETTINGS,
+    type: TOGGLE_SETTINGS,
+    payload: showSettings,
   });
 
-const closeSettings = () => (dispatch) =>
+const toggleLoader = (showLoader) => (dispatch) =>
```
Comment (juancarlosfarah) [inline src/actions/layout.js]: This is for the `ProgressScreen`, right? Maybe gets confused with the `Loader` component. Let's be consistent. I think renaming the component to `LoadingScreen` and then using `toggleLoadingScreen` and `showLoadingScreen` would be clearer.

File: src/components/App.js [hunk 2]
```diff
@@ -66,48 +67,42 @@ export class App extends Component {
...
+      case MODES.PRODUCER:
+      case MODES.EDUCATOR:
+      case MODES.ADMIN:
+        return <TeacherMode view={view} />;
 
       // by default go with the consumer (learner) mode
-      case 'student':
-      case 'consumer':
-      case 'learner':
+      case MODES.STUDENT:
```
Comment (juancarlosfarah) [inline src/components/App.js:89]: Love this.

**Comment by juancarlosfarah** [comment]
What does the purple button on the top left do?

File: src/components/common/Main.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    alignItems: 'center',
+    padding: theme.spacing(0, 1),
+    ...theme.mixins.toolbar,
+    justifyContent: 'flex-end',
+  },
+  fab: {
+    left: theme.spacing(2),
+    top: theme.spacing(2),
+    position: 'absolute',
+    zIndex: 999999,
```
Comment (juancarlosfarah) [inline src/components/common/Main.js]: Can we factor this out to a constant? Like `MAXIMUM_Z_INDEX` or something like that.

File: src/components/modes/teacher/TeacherView.js [hunk 4]
```diff
@@ -1,282 +1,85 @@
...
+const styles = (theme) => ({
+  message: {
+    margin: theme.spacing(2),
+    padding: theme.spacing(1),
+    textAlign: 'center',
+    backgroundColor: theme.status.danger.background[500],
+    color: theme.status.danger.color,
+  },
+  fab: {
+    zIndex: 10000,
```
Comment (juancarlosfarah) [inline src/components/modes/teacher/TeacherView.js]: Can we factor this out like above? Not sure what to call it, though. `FAB_Z_INDEX`?

File: src/config/i18n.js [hunk 5]
```diff
@@ -23,4 +23,11 @@ i18n.use(initReactI18next).init({
   },
 });
 
+const langs = {
+  en: 'English',
+  fr: 'Français',
```
Comment (juancarlosfarah) [inline src/config/i18n.js:28]: Very nice! I like this.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks excellent! Just some very minor changes.

**Review by hagoptaminian — APPROVED** [review]
this looks good to me. I know we still have to build out `README.md` but might be worth already adding a note like:

_Once you clone the repository, to start the application correctly make sure to run_ `touch .env.local` _and_ `echo "REACT_APP_BASE=" >> .env.local` _in the project directory._

But of course could be done later in a README.md PR

**Review by juancarlosfarah — APPROVED** [review]

**Comment by pyphilia** [comment]
I updated the readme with some steps to make the application run, and a tiny refactor for the buttons. I will merge it.

**Pull request merged**